src/utils/embedded/utils/messages/embedded-messages.utils.ts

```diff
@@ -0,0 +1,45 @@
+import { nanoid } from "nanoid";
+import { EMBEDDED_PARENT_ORIGIN } from "~utils/embedded/sdk/utils/url/sdk-url.utils";
+import type {
+  EmbeddedCall,
+  EmbeddedMessageId,
+  EmbeddedMessageMap
+} from "~utils/embedded/utils/messages/embedded-messages.types";
+
+export interface PostEmbeddedMessageData<K extends EmbeddedMessageId> {
+  type: K;
+  data: EmbeddedMessageMap[K];
+}
+
+export function postEmbeddedMessage<K extends EmbeddedMessageId>({
+  type,
+  data
+}: PostEmbeddedMessageData<K>) {
+  if (!type.startsWith("embedded_"))
+    throw new Error(
+      `Only "embedded_auth", "embedded_balance", "embedded_resize" or "embedded_data" can be posted to the upper frame.`
```

Comment:
`embedded_data` is this intentional ? embedded_close is missing i guess based on map in types file

Reply:
My bad, the last should be `embedded_close`. I've pushed commit with the correct code.